core/LSU.cpp

```diff
@@ -452,12 +679,218 @@ namespace olympia
         // instruction issue arbitration should always succeed, even when flush happens.
         // Otherwise, assertion error is fired inside arbitrateInstIssue_()
     }
+    void LSU::replayReady_(const InstPtr & replay_inst_ptr)
+    {
+        ILOG("Replay inst ready " << replay_inst_ptr);
+        // We check in the ldst_queue as the instruction may not be in the replay queue
+        for (auto & load_store_info_ptr : ldst_inst_queue_)
+        {
+            auto & inst_ptr = load_store_info_ptr->getInstPtr();
+            if (inst_ptr == replay_inst_ptr)
+            {
+                if(load_store_info_ptr->getState() == LoadStoreInstInfo::IssueState::NOT_READY)
+                {
+                    load_store_info_ptr->setState(LoadStoreInstInfo::IssueState::READY);
+                }
+                auto issue_priority = load_store_info_ptr->getMemoryAccessInfoPtr()->getPhyAddrStatus()
+                                          ? LoadStoreInstInfo::IssuePriority::CACHE_PENDING
+                                          : LoadStoreInstInfo::IssuePriority::MMU_PENDING;
+                load_store_info_ptr->setPriority(issue_priority);
+            }
+        }
+        if (isReadyToIssueInsts_())
+        {
+            ILOG("replay ready issue");
+            uev_issue_inst_.schedule(sparta::Clock::Cycle(0));
+        }
+    }
+
+    void LSU::updateInstReplayReady(const InstPtr &inst_ptr){
+        for(auto iter = replay_buffer_.begin(); iter != replay_buffer_.end(); iter++)
+        {
+            auto &load_store_info_ptr = *iter;
+            if (load_store_info_ptr->getInstPtr() == inst_ptr)
+            {
+                ILOG("Scheduled replay " << load_store_info_ptr << " after " << replay_issue_delay_ << " cycles");
+                load_store_info_ptr->setState(LoadStoreInstInfo::IssueState::NOT_READY);
+                uev_replay_ready_.preparePayload(inst_ptr)->schedule(sparta::Clock::Cycle(replay_issue_delay_));
+                replay_buffer_.erase(iter);
+
+                replay_insts_++;
+                return;
+            }
+        }
 
+        sparta_assert(false, "Cannot replay a non existing instruction " << inst_ptr);
+    }
 
     ////////////////////////////////////////////////////////////////////////////////
     // Regular Function/Subroutine Call
     ////////////////////////////////////////////////////////////////////////////////
 
+    bool LSU::instPresentInQueues(const InstPtr &inst_ptr)
```

Comment:
Please add documentation (even if just one line) stating what each method is doing.

Reply:
Yes I will add them